src/libponyrt/lang/stat.c

```diff
@@ -69,6 +71,8 @@ static void windows_stat(const char* path, pony_stat_t* p, struct __stat64* st,
   p->broken = false;
 
   p->hard_links = (uint32_t)st->st_nlink;
+  p->device = (uint32_t)st->st_dev;
```

Comment:
I figured out the different types using `clang -E test.c | grep <TYPE>` or `gcc -E test.c | grep <TYPE>` and following the chain of types to get to the underlying machine type. `test.c` included every posix header file and declared a variable of each type. I did this on both OSX and Linux (ubuntu in a VM) both on my laptop.